packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -899,12 +671,165 @@ export const prepareVaultManagerKit = (
+
+          /** @type {import('./storeUtils.js').NormalizedDebt} */
+          // @ts-expect-error cast
+          const crKey = normalizedCrKey(lockedQuote, compoundedInterest);
+
+          trace(
+            `Liquidating vaults worse than`,
+            normalizedCr(lockedQuote, compoundedInterest),
+          );
+          const { totalDebt, totalCollateral, vaultsToLiquidate, liqSeat } =
+            prioritizedVaults.removeVaultsBelow(crKey, zcf);
+          if (!totalCollateral) {
+            return;
+          }
+
+          vaultsToLiquidate.forEach(([_, v]) => {
+            v.liquidating();
+            liquidatingVaults.add(v);
+          });
+
+          trace(' Found vaults to liquidate', liquidatingVaults.getSize());
+
+          const { userSeatPromise, deposited } = await E.when(
+            E(auctionPF).getDepositInvitation(),
+            depositInvitation =>
+              offerTo(
+                zcf,
+                depositInvitation,
+                harden({ Minted: 'Currency' }),
+                harden({
+                  give: { Collateral: totalCollateral },
+                }),
+                liqSeat,
+              ),
+          );
+
+          // This can be a long wait.
+          const [proceeds] = await Promise.all([deposited, userSeatPromise]);
+
+          trace(`VM LiqV after long wait`, proceeds);
+
+          const accounting = liquidationResults(
+            totalDebt,
+            proceeds.Minted || AmountMath.makeEmpty(debtBrand),
+          );
+
+          const removeLiquidatedVault = v => {
+            trace(`removing a liquidated vault`);
+            v.liquidated();
+            liquidatingVaults.delete(v);
+          };
+
+          if (AmountMath.isEmpty(accounting.overage)) {
+            // no one gets anything back
+            facets.manager.burnAndRecord(accounting.toBurn, liqSeat);
+            if (
+              !AmountMath.isEmpty(proceeds.Collateral) &&
+              !AmountMath.isEmpty(accounting.shortfall)
+            ) {
+              // The borrower doesn't get the excess collateral remaining when
+              // liquidation results in a shortfall.
+              // TODO  transfer it to the reserve.
+              atomicTransfer(zcf, liqSeat, retainedCollateralSeat, {
+                Collateral: proceeds.Collateral,
+              });
+            }
+
+            updateStateFromAccounting(state, accounting, proceeds.Collateral);
+            facets.helper.updateMetrics();
+
+            if (!AmountMath.isEmpty(accounting.shortfall)) {
+              E(factoryPowers.getShortfallReporter())
+                .increaseLiquidationShortfall(accounting.shortfall)
+                .catch(reason =>
+                  console.error(
+                    'liquidVaults failed to increaseLiquidationShortfall',
+                    reason,
+                  ),
+                );
+            }
+
+            vaultsToLiquidate.forEach(([_, v]) => removeLiquidatedVault(v));
```

Comment:
`for` over `.values()`. though I can see an argument for:
```suggestion
            Object.values(vaultsToLiquidate).forEach(removeLiquidatedVault);
```

though why is `vaultsToLiquidate ` a list of entries instead of a Map?

Reply:
converted to a Map.  The types don't propagate well.